app/services/unsubscribe_service.rb

```diff
@@ -1,37 +1,52 @@
 module UnsubscribeService
   class << self
     def subscriber!(subscriber, reason)
-      unsubscribe!(subscriber, subscriber.active_subscriptions, reason)
+      ActiveRecord::Base.transaction do
+        unsubscribe_subscriptions!(subscriber.active_subscriptions, reason)
+        unsubscribe_subscriber!(subscriber)
+      end
     end
 
     def subscription!(subscription, reason)
-      unsubscribe!(subscription.subscriber, [subscription], reason)
+      ActiveRecord::Base.transaction do
+        unsubscribe_subscriptions!([subscription], reason)
+        unsubscribe_subscriber!(subscription.subscriber)
+      end
     end
 
     def spam_report!(delivery_attempt)
       subscriber_id = delivery_attempt.email.subscriber_id
       subscriber = Subscriber.find(subscriber_id)
-      unsubscribe!(subscriber, subscriber.active_subscriptions, :marked_as_spam, delivery_attempt.email)
+      ActiveRecord::Base.transaction do
+        unsubscribe_subscriptions!(subscriber.active_subscriptions, :marked_as_spam, delivery_attempt.email)
+        unsubscribe_subscriber!(subscriber)
+      end
     end
 
-  private
-
-    def unsubscribe!(subscriber, subscriptions, reason, email = nil)
```

Comment:
I'm a bit unclear what part of this commit is refactoring, and what purpose the refactoring is for. It would be useful to have some more information in the commit message, or maybe either separate the refactoring out in to a different commit.